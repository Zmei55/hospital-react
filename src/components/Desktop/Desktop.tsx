import {
  Container,
  LargeContainer,
  MediumContainer,
  SmallContainer,
  DesktopLink,
  IconContainer,
  IconTitle,
  IconNewRequest,
  IconFindRequest,
  IconFindContainer,
  IconDocuments,
} from './Desktop.styled';

export const Desktop: React.FC = () => {
  return (
    <Container>
      <LargeContainer>
        <DesktopLink to="/patient">
          <IconContainer>
            <IconNewRequest />
            <IconTitle>Erstellen Sie eine Bestellung</IconTitle>
          </IconContainer>
        </DesktopLink>
      </LargeContainer>

      <MediumContainer>
        <DesktopLink to={''}>
          <IconContainer>
            <IconFindRequest />
            <IconTitle>Finden Sie eine Bestellung</IconTitle>
          </IconContainer>
        </DesktopLink>
      </MediumContainer>

      <MediumContainer>
        <DesktopLink to={''}>
          <IconContainer>
            <IconFindContainer />
            <IconTitle>Finden Sie eine BehÃ¤lter</IconTitle>
          </IconContainer>
        </DesktopLink>
      </MediumContainer>

      <SmallContainer>
        <DesktopLink to={''}>
          <IconContainer>
            <IconDocuments />
            <IconTitle>Unterlagen</IconTitle>
          </IconContainer>
        </DesktopLink>
      </SmallContainer>
    </Container>
  );
};
