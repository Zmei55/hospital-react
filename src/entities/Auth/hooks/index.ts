export * from "./useLogIn";
export * from "./useLogOut";
export * from "./useHandleChange";
