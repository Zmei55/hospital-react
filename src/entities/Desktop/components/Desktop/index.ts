export * from "./Desktop";
export { Desktop as default } from "./Desktop";
