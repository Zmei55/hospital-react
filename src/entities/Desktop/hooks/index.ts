export * from "./useNewRequest";
export * from "./useFindRequest";
