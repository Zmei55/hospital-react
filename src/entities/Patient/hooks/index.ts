export * from "./useChoosePatient";
export * from "./usePatientListFind";
export * from "./useHandleInfoActive";
export * from "./useClearPatient";
export * from "./useTogglePatientsModal";
