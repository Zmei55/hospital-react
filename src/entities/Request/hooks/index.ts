export * from "./useRequestServices";
export * from "./useSendRequestDetails";
