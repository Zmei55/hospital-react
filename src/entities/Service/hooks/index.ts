export * from "./useServiceAdd";
export * from "./useServiceDelete";
export * from "./useSelectedServiceDelete";
export * from "./useSelectedServiceListSave";
export * from "./useSelectedServiceListClear";
export * from "./useGetLaborsList";
export * from "./useGetServiceById";
export * from "./useGetServiceListById";
export * from "./useHandleLaborChange";
export * from "./useHandleServicesForm";
export * from "./useToggleLaborModal";
export * from "./useToggleServicesModal";
