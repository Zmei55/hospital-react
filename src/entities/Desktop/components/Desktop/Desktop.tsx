import { useTranslation } from "react-i18next";
import { useNewRequest, useFindRequest } from "entities/Desktop";
import { useAppNavigate, Icon } from "shared";

import {
  Container,
  LargeBtn,
  MediumBtn,
  SmallBtn,
  IconContainer,
  IconTitle,
} from "./Desktop.styled";

export const Desktop: React.FC = () => {
  const [navigate] = useAppNavigate();
  const { handleNewRequest } = useNewRequest();
  const { handleFindRequest } = useFindRequest();
  const { t } = useTranslation();

  return (
    <Container data-testid="desktop-page">
      <LargeBtn data-testid="newRequestBtn" onClick={handleNewRequest}>
        <IconContainer>
          <Icon icon="create-document" color="blue" size={140} />
          <IconTitle>{t("desktop.addRequest")}</IconTitle>
        </IconContainer>
      </LargeBtn>

      <MediumBtn data-testid="findRequestBtn" onClick={handleFindRequest}>
        <IconContainer>
          <Icon icon="find-document" color="blue" size={140} />
          <IconTitle>{t("desktop.findRequest")}</IconTitle>
        </IconContainer>
      </MediumBtn>

      <MediumBtn
        data-testid="findContainersBtn"
        onClick={() => navigate("/unknown-part")}
      >
        <IconContainer>
          <Icon icon="find-test-tube" color="blue" size={140} />
          <IconTitle>{t("desktop.findContainer")}</IconTitle>
        </IconContainer>
      </MediumBtn>

      <SmallBtn
        data-testid="documentsBtn"
        onClick={() => navigate("/unknown-part")}
      >
        <IconContainer>
          <Icon icon="documents" color="blue" size={140} />
          <IconTitle>{t("desktop.documents")}</IconTitle>
        </IconContainer>
      </SmallBtn>
    </Container>
  );
};
