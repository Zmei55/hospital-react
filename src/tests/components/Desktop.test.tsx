import { render, screen } from "@testing-library/react";
import { allTheProviders } from "tests/utils";
import { Desktop } from "entities/Desktop";
import * as sharedHooks from "shared/hooks";

jest.mock("shared/hooks");

const mockedUseNavigate = jest.spyOn(sharedHooks, "useAppNavigate");

describe("Desktop component", () => {
  it("Desktop renders", () => {
    mockedUseNavigate.mockReturnValue({ handleNavigate: jest.fn() });

    const { container } = render(allTheProviders(<Desktop />));

    expect(screen.getByTestId("desktop-page")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("all components renders", () => {
    mockedUseNavigate.mockReturnValue({ handleNavigate: jest.fn() });

    render(allTheProviders(<Desktop />));

    const newRequestBtn = screen.getByTestId("new-request-btn");
    const findRequestBtn = screen.getByTestId("find-request-btn");
    const findContainersBtn = screen.getByTestId("find-containers-btn");
    const documentsBtn = screen.getByTestId("documents-btn");

    expect(newRequestBtn).toBeInTheDocument();
    expect(screen.getByText(/create a request/i)).toBeInTheDocument();

    expect(findRequestBtn).toBeInTheDocument();
    expect(screen.getByText(/find a request/i)).toBeInTheDocument();

    expect(findContainersBtn).toBeInTheDocument();
    expect(screen.getByText(/find a container/i)).toBeInTheDocument();

    expect(documentsBtn).toBeInTheDocument();
    expect(screen.getByText(/documents/i)).toBeInTheDocument();
  });

  it("all navigates are called", () => {
    const handleNavigate = jest.fn();

    mockedUseNavigate.mockReturnValue({
      handleNavigate: handleNavigate("/request"),
    });
    mockedUseNavigate.mockReturnValue({
      handleNavigate: handleNavigate("/find-request"),
    });
    mockedUseNavigate.mockReturnValue({
      handleNavigate: handleNavigate("/unknown-part"),
    });

    render(allTheProviders(<Desktop />));

    expect(handleNavigate).toHaveBeenCalledTimes(3);
  });
});
