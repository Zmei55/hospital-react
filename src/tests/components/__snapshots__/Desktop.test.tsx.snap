// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Desktop component Desktop renders 1`] = `
<div>
  <div
    class="css-5n3p1z"
    data-testid="desktop-page"
  >
    <button
      class="css-wqedcp"
      data-testid="new-request-btn"
    >
      <div
        class="css-dv5uw6"
      >
        <svg
          class="css-v61enk"
          fill="blue"
          height="140"
          width="140"
        >
          <use
            href="svgSprite.svg#create-document"
          />
        </svg>
        <span
          class="css-9j55n3"
        >
          Create a request
        </span>
      </div>
    </button>
    <button
      class="css-8ra3jc"
      data-testid="find-request-btn"
    >
      <div
        class="css-dv5uw6"
      >
        <svg
          class="css-v61enk"
          fill="blue"
          height="140"
          width="140"
        >
          <use
            href="svgSprite.svg#find-document"
          />
        </svg>
        <span
          class="css-9j55n3"
        >
          Find a request
        </span>
      </div>
    </button>
    <button
      class="css-8ra3jc"
      data-testid="find-containers-btn"
    >
      <div
        class="css-dv5uw6"
      >
        <svg
          class="css-v61enk"
          fill="blue"
          height="140"
          width="140"
        >
          <use
            href="svgSprite.svg#find-test-tube"
          />
        </svg>
        <span
          class="css-9j55n3"
        >
          Find a container
        </span>
      </div>
    </button>
    <button
      class="css-1l4ikde"
      data-testid="documents-btn"
    >
      <div
        class="css-dv5uw6"
      >
        <svg
          class="css-v61enk"
          fill="blue"
          height="140"
          width="140"
        >
          <use
            href="svgSprite.svg#documents"
          />
        </svg>
        <span
          class="css-9j55n3"
        >
          Documents
        </span>
      </div>
    </button>
  </div>
</div>
`;
